### ISSUE TITLE
Improve "update native messaging manifest" advanced option description

### ISSUE THREAD

**Issue by TheAssassin** [keepassxreboot/keepassxc#3315]
The message highlighted in @droidmonkey's screenshot is a bit hard to grasp for non-experts:

![](https://user-images.githubusercontent.com/2809491/59984963-62006100-95fc-11e9-8560-1c4122e910ea.png)

I had no idea what native messaging even means before I discovered the solution for the bug in #3314. The message should IMO at least mention what difference this makes.

Ideally, I'd like to see this become default behavior and remove the option at all.

**Comment by varjolintu** [comment]
It's already the default behaviour. This option is still needed because development without it would be much harder. With this option disabled it's possible to use a dev/test version of the proxy without KeePassXC overriding the scripts every time it's started.

**Comment by TheAssassin** [comment]
> It's already the default behaviour

I disagree. I even tested this quickly, and the AppImage does *not* have that enabled by default.

> This option is still needed because development without it would be much harder. With this option disabled it's possible to use a dev/test version of the proxy without KeePassXC overriding the scripts every time it's started.

Makes sense. Thanks for the clarification.

**Comment by varjolintu** [comment]
> I disagree. I even tested this quickly, and the AppImage does _not_ have that enabled by default.

Well, that's strange because https://github.com/keepassxreboot/keepassxc/pull/2763 fixed it.

**Comment by TheAssassin** [comment]
I guess my KeePassXC version is too old then, let me re-try with the latest AppImage (not on my main computer right now).

**Comment by TheAssassin** [comment]
You're right @varjolintu. I guess this bug then only affects users who already had some config files, if you run w/o preexisting configs this is default.

You might want to make one intermediate release that force-changes this option to be switched on. That might be a little inconvenient to you developers, since you then have to take action again, but for the users it's great.

**Comment by phoerious** [comment]
I'm actually not sure if we ever had it off by default or of that only happens to be in your config for whatever reason.

**Comment by TheAssassin** [comment]
I never turned it off, and the behavior is reproducible on older KeePassXC versions like KeePassXC 2.3.

**Comment by phoerious** [comment]
Ok, then we should perhaps push an update for this. But it's quite risky, especially for all those users who use custom manifests for their not officially supported browsers. This shouldn't be done without user interaction. Or at least only update manifests if the executable path points to somewhere else, but leave configured browser extension IDs untouched.

**Comment by droidmonkey** [comment]
I am against forcing the change on an update.

**Comment by TheAssassin** [comment]
A little more complexity to the rescue: you could add a tiny additional config setting that is not shown anywhere like "performed auto-switch on already", which is set on initially, and is set to on after having changed the setting automatically after the update. That setting may never go away again in the future, and if it's there, the config will not be auto-changed any more in the future, which reduces the disturbance for existing developers who even ever touched that setting to having to switch it on once again after that update, and never worry about it in the future again.
This can be modelled either as a separate setting, or the information is embedded in the existing setting by introducing a third state "on (auto setting)".

But I see there might be too much effort in this. However, it's annoying for all existing AppImage users to basically have a broken config. So, if there aren't any code changes, there really has to be some documentation on this bug that clearly relates this setting to the bug when KeePassXC-Browser shows this weird message.

I also think that it might help if KeePassXC-Browser could just say "could not find file XXXXXXX on disk" or so, instead of "key exchange failed".

**Comment by varjolintu** [comment]
The simplest solution would be to change the internal name of the option and enable it by default.

@TheAssassin The problem with KeePassXC-Browser errors is that Native Messaging API doesn't trigger certain errors at all. This is why a same error might be displayed in multiple situations.

**Comment by droidmonkey** [comment]
Making the setting more visible and with better wording is the best fix.

**Comment by TheAssassin** [comment]
Perhaps this can be documented somewhere, worded accordingly. Is it possible to put a popup there in the app? That'll also help clarify what the option is about, and what it's for. I can put together a few sentences.

**Comment by droidmonkey** [comment]
Yes you can add a _tooltip_ to the checkbox that can show additional information when you hover your mouse over it.
